shopify-features/models/profiles.yaml

```diff
@@ -294,37 +330,40 @@ models:
             name: active_days_in_past_7_days
             select: count(distinct date(session_start_time))
             from: models/rsSessionTable
-            where: "{% if !(end_time|isnil) %}datediff(day,date(session_start_time),date('{{end_time.Format(\"2006-01-02 15:04:05\")}}')) <= 7{% else %} 1=1 {% endif %}"
+            where: "{{macro_datediff('session_start_time')}} <=7"
             description: out of 7 days, how many days have recorded an event till date including today
         - entity_var:
             name: active_days_in_past_365_days
             select: count(distinct date(session_start_time))
             from: models/rsSessionTable
-            where: "{% if !(end_time|isnil) %}datediff(day,date(session_start_time),date('{{end_time.Format(\"2006-01-02 15:04:05\")}}')) <= 365{% else %} 1=1 {% endif %}"
+            # where: "{% if !(end_time|isnil) %}datediff(day,date(session_start_time),date('{{end_time.Format(\"2006-01-02 15:04:05\")}}')) <= 365{% else %} 1=1 {% endif %}"
+            where: "{{macro_datediff('session_start_time')}} <=365"
+
             description: out of 365 days, how many days have recorded an event till date including todays date
 
         - entity_var:
             name: total_sessions_till_date
-            from: models/rsSessionTable
-            select: count(*)
-            description: Total individual sessions created till date.
+            from: models/rsTracksUnionPages
```

Comment:
Keep this to rsSessionTable. That was optimised with only session start and end events.